js/lightbeam.js

```diff
@@ -20,6 +24,82 @@ const lightbeam = {
     });
   },
 
+  // Called from init() (isFirstParty = undefined)
+  // and redraw() (isFirstParty = true or false).
+  async updateVars(isFirstParty) {
+
+    // initialize dynamic vars from storage
+    if (!this.dataGatheredSince) {
+      const { dateStr, fullDateTime } = await this.getDataGatheredSince();
+      this.dataGatheredSince = dateStr;
+      const dataGatheredSince = document.getElementById('data-gathered-since');
+      dataGatheredSince.textContent = this.dataGatheredSince;
+      dataGatheredSince.setAttribute('datetime', fullDateTime);
+    }
+    if (isFirstParty === undefined) {
+      this.numFirstParties = await this.getNumFirstParties();
+      this.setPartyVar(true);
+      this.numThirdParties = await this.getNumThirdParties();
```

Comment:
you can ignore this comment, but can we change `numFirstParties` to `firstPartyCount` ? :)